File: AdSurgeDemo/AdSurgeDemo/AdSurgeDemoUI/BannerViewController.m
//
//  BannerViewController.m
//  AdSurgeDemo
//
//  Created by admin on 2025/9/8.
//

#import <Foundation/Foundation.h>

#import <AdSurgeSDK/AdSurgeBannerAdView.h>

#import "AdLogTable.h"
#import "BannerViewController.h"

@interface BannerViewController ()<AdSurgeBannerAdDelegate>

@property (nonatomic, strong) UIView *blackBackgroundView;
@property (nonatomic) AdSurgeBannerAdView *bannerView;
@property (nonatomic, strong) AdLogTable *logTable;

@end

@implementation BannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self removeShowButton];
    NSMutableArray *unitIds = [NSMutableArray arrayWithArray:@[
        @"15884",
        @"15877",
    ]];
    NSMutableArray *titles = [NSMutableArray arrayWithArray:@[
        @"Banner 15884",
        @"MREC 15877",
    ]];
    self.logTable = [[AdLogTable alloc] initWithParentView:self.view withTopMargin:250];
    self.unitIds = unitIds;
    self.unitIdTitles = titles;
    self.unitId = self.unitIds.firstObject;
}

- (void)loadAd {
    if (self.bannerView.superview) {
        [self.bannerView removeFromSuperview];
        self.bannerView = nil;
    }
    AdSurgeBannerAdView *bannerView = [[AdSurgeBannerAdView alloc] initWithAdUnitIdentifier:self.unitId];
    bannerView.viewController = self;
    bannerView.delegate = self;
    self.bannerView = bannerView;
    CGRect rect;
    rect = CGRectMake(10, 200, 320, 50);
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
        rect = CGRectMake(10, 200, 728, 90);
    }
    if (self.bannerView) {
        self.bannerView.frame = rect;
    }
    [self.view addSubview:self.bannerView];
    
    AdSurgeAdConfig *adConfig = [AdSurgeAdConfig configWithAdFormat:AdSurgeAdFormat.banner];
    [self.bannerView loadAd:adConfig];

}



#pragma mark - AdSurgeRewardedAdDelegate

- (void)didLoadAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"ad's format: %@, adUnitIdentifier: %@, revenue: %f",ad.format.label, ad.adUnitIdentifier, ad.revenue);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(AdSurgeError *)error {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"ad's adUnitIdentifier: %@, error: %@",adUnitIdentifier, error);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:nil error:error reward:nil];
}

- (void)didHideAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didClickAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didPayRevenueForAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}


@end

File: AdSurgeDemo-Swift/AdSurgeDemo-Swift/AdSurgeDemoUI/BannerViewController.swift
//
//  BannerViewController.swift
//  AdSurgeDemo-Swift
//
//  Created by admin on 2025/9/8.
//

import UIKit
import AdSurgeSDK
import Foundation

class BannerViewController: BaseAdViewController, AdSurgeBannerAdDelegate {

    var bannerView: AdSurgeBannerAdView!
    override func viewDidLoad() {
        super.viewDidLoad()
        self.unitIdTitles = [
            "Banner 15884",
            "MREC 15877"
        ];
        self.unitIds = [
            "15884",
            "15877"
        ]
        self.unitId = unitIds.first
        removeShowButton()
    }

    override func loadAd() {
        if let banner = bannerView, banner.superview != nil {
        banner.removeFromSuperview()
        bannerView = nil
        }
        let localBanner = AdSurgeBannerAdView(adUnitIdentifier: unitId ?? "15884")
        localBanner.viewController = self
        localBanner.delegate = self
        self.bannerView = localBanner

        var rect = CGRect(x: 10, y: 200, width: 320, height: 50)
        if UIDevice.current.userInterfaceIdiom == .pad {
            rect = CGRect(x: 10, y: 200, width: 728, height: 90)
        }
        self.bannerView.frame = rect

        if let banner = self.bannerView {
            view.addSubview(banner)
        }

        let adConfig = AdSurgeAdConfig(adFormat: .banner)
        bannerView.loadAd(adConfig)
    }
    
    // MARK: - AdSurgeRewardedAdDelegate

    func didLoad(_ ad: AdSurgeAd) {
        print(#function)
        print("ad's format: \(ad.format.label), adUnitIdentifier: \(ad.adUnitIdentifier), revenue: \(ad.revenue)")
    }

    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: AdSurgeError) {
        print(#function)
        print("ad's adUnitIdentifier: \(adUnitIdentifier), error: \(error)")
    }

    func didHide(_ ad: AdSurgeAd) {
        print(#function)
    }

    func didClick(_ ad: AdSurgeAd) {
        print(#function)
    }

    func didPayRevenue(for ad: AdSurgeAd) {
        print(#function)
    }
}
